sdk/search/search-documents/src/searchIndexingBufferedSender.ts

```diff
@@ -0,0 +1,351 @@
+import { SearchClient } from "./searchClient";
+import { IndexDocumentsBatch } from "./indexDocumentsBatch";
+import {
+  IndexDocumentsAction,
+  SearchIndexingBufferedSenderOptions,
+  SearchIndexingBufferedSenderUploadDocumentsOptions,
+  SearchIndexingBufferedSenderMergeDocumentsOptions,
+  SearchIndexingBufferedSenderMergeOrUploadDocumentsOptions,
+  SearchIndexingBufferedSenderDeleteDocumentsOptions,
+  SearchIndexingBufferedSenderFlushDocumentsOptions
+} from "./indexModels";
+import { IndexDocumentsResult } from "./generated/data/models";
+import { RestError, OperationOptions } from "@azure/core-http";
+import EventEmitter from "events";
+import { createSpan } from "./tracing";
+import { CanonicalCode } from "@opentelemetry/api";
+
+const DEFAULT_BATCH_SIZE: number = 1000;
+const DEFAULT_FLUSH_WINDOW: number = 60000;
+const RETRY_COUNT: number = 3;
+
+/**
+ * Class used to perform buffered operations against a search index,
+ * including adding, updating, and removing them.
+ */
+export class SearchIndexingBufferedSender<T> {
+  /**
+   * Search Client used to call the underlying IndexBatch operations.
+   */
+  private client: SearchClient<T>;
+  /**
+   * Indicates if autoFlush is enabled.
+   */
+  private autoFlush: boolean;
+  /**
+   * Interval between flushes (in milliseconds).
+   */
+  private flushWindowInMs: number;
+  /**
+   * Size of the batch.
+   */
+  private batchSize: number;
+  /**
+   * Batch object used to complete the service call.
+   */
+  private batchObject: IndexDocumentsBatch<T>;
+  /**
+   * Clean up for the timer
+   */
+  private cleanupTimer?: () => void;
+  /**
+   * Event emitter/publisher used in the Buffered Sender
+   */
+  private readonly emitter = new EventEmitter();
+
+  /**
+   * Creates a new instance of SearchIndexingBufferedSender.
+   *
+   * @param client Search Client used to call the underlying IndexBatch operations.
+   * @param options Options to modify batch size, auto flush and flush window.
```

Comment:
this comment is out of date, since you can't customize anything but auto flush

Reply:
Fixed